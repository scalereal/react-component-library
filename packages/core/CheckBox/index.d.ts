import { FC } from 'react';
import { WithStyle } from '../../utils';
import { Props } from './types';
declare const CheckBox: FC<Props> & WithStyle;
export default CheckBox;
