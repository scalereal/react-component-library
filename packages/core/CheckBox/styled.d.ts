import { Props } from './types';
export declare const mixin: (size: string) => "input[type=\"checkbox\"]{\n                width: 1.4rem;\n                height: 1.4rem;\n                }\n                input[type=\"checkbox\"]:checked:before {\n                    width: 4px;\n                    height: 10px;\n                    border-width: 0 3px 3px 0;\n                    margin-top: 2px;\n                }\n            " | "input{\n                width: 1.8rem;\n                height: 1.8rem;\n                }\n                input[type=\"checkbox\"]:checked:before {\n                    width: 5px;\n                    height: 14px;\n                    border-width: 0 3px 3px 0;\n                    margin-top: 4px;\n                }\n            " | "input{\n                width: 2.2rem;\n                height: 2.2rem;\n                }\n                input[type=\"checkbox\"]:checked:before {\n                    width: 7px;\n                    height: 18px;\n                    border-width: 0 4px 4px 0;\n                    margin-top: 4px;\n                }\n            " | "input{\n                width: 2.6rem;\n                height: 2.6rem;\n                }\n                input[type=\"checkbox\"]:checked:before {\n                    width: 7px;\n                    height: 21px;\n                    border-width: 0 5px 5px 0;\n                    margin-top: 4px;\n                }\n            " | "input{\n                width: 3rem;\n                height: 3rem;\n                }\n                input[type=\"checkbox\"]:checked:before {\n                    width: 8px;\n                    height: 24px;\n                    border-width: 0 5.5px 5.5px 0;\n                    margin-top: 4px;\n                }\n            " | undefined;
export declare const ErrorLabel: import("styled-components").StyledComponent<"label", any, Props, never>;
export declare const StyledLabel: import("styled-components").StyledComponent<"label", any, Props, never>;
export declare const CheckBoxWrapper: import("styled-components").StyledComponent<"div", any, Props, never>;
