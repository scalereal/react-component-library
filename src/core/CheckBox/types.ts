
export type CheckBoxSizes = 'XS' | 'S' | 'M' | 'L' | 'XL' ;
export type LabelPositions = 'left'| 'right'| 'top'| 'bottom';

export interface Props{
    size?:CheckBoxSizes;
    label?:string;
    value?:string|number|any;
    error?:boolean;
    bgColor?:string;
    labelColor?:string;
    hoverEffect?:boolean;
    hoverLabelColor?:string;
    hoverBgColor?:string;
    hoverShadowColor?:string;
    hoverLabelShadowColor?:string;
    disabled?:boolean;
    fullWidth?: boolean;
    /** Label position */
    labelPosition?: LabelPositions;
    /** Label Variant */
    labelVariant?: string;
    /** Label Weight */
    labelWeight?: string;
    /** Indeterminate state for checkbox */
    indeterminate?: boolean;
    /** Error Text */
    errorText?: string;
    /** Function will called with the input value on blur, change and invalid event */
    validator?: (checked: boolean) => string;
    /** This will be used internally by Checkbox group */
    hasError?: boolean;
}