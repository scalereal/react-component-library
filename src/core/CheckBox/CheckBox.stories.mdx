import CheckBox, { lposMap, SizeMap } from '.';
import { Preview, Story, Meta, Props, ArgsTable, Canvas } from '@storybook/addon-docs/blocks';
import { withKnobs, color, boolean, select } from '@storybook/addon-knobs';

<Meta
    title="Components/Core/CheckBox"
    component={CheckBox}
    argTypes={{
        size: {
            control: {
                type: 'select',
                options: Object.values(SizeMap)
            }
        },
        fontSize: {
            control: {
                type: 'select',
                options: Object.values(SizeMap)
            }
        },
        labelPosition: {
            control: {
                type: 'select',
                options: Object.values(lposMap)
            }
        },
        label: {
            control: { type: 'text' }
        },
        value: {
            control: { type: 'text' }
        },
        hoverEffect: {
            control: { type: 'boolean' }
        },
        labelHoverEffect: {
            control: { type: 'boolean' }
        },
        disabled: {
            control: { type: 'boolean' }
        },
        hasError: {
            control: { type: 'boolean' }
        },
        errorText: {
            control: { type: 'text' }
        },
        bgColor: {
            control: 'color'
        },
        labelColor: {
            control: 'color'
        },
        hoverBgColor: {
            control: 'color'
        },
        labelHoverColor: {
            control: 'color'
        },
        hoverShadowColor: {
            control: 'color'
        },
        ref: {
            table: {
                disable: true
            }
        },
        theme: {
            table: {
                disable: true
            }
        }
    }}
/>

# CheckBox

Displays a CheckBox.

export const Template = args => (
    <div>
        <CheckBox {...args} />
    </div>
);

<Preview>
    <Story name="CheckBox" args={{ size: 'S' }}>
        {Template.bind({})}
    </Story>
</Preview>

### Color Variations

You can either change the theme or pass the colors as props.

<Preview withToolbar>
    <div>
        <CheckBox checked={true} bgColor="#37d5d3" />
        <CheckBox checked={true} bgColor="#2a0481" />
        <CheckBox checked={true} bgColor="#37d5d3" labelColor="blue" />
        <CheckBox checked={true} bgColor="#2a0481" labelColor="red" />
    </div>
</Preview>

### Error

Displays Error Text if hasError={true}.

<Preview withToolbar>
    <div>
        <CheckBox hasError={true} errorText="Please check the CheckBox" />
    </div>
</Preview>

### Sizes

There are five sizs available `XS` | `S` | `M` | `L` | `XL`.

<Preview withToolbar>
    <div>
        <CheckBox size="XS" />
        <CheckBox size="S" />
        <CheckBox size="M" />
        <CheckBox size="L" />
        <CheckBox size="XL" />
    </div>
</Preview>

### Text Sizes

There are five Text sizs available `XS` | `S` | `M` | `L` | `XL`.

<Preview withToolbar>
    <div>
        <CheckBox fontSize="XS" />
        <CheckBox fontSize="S" />
        <CheckBox fontSize="M" />
        <CheckBox fontSize="L" />
        <CheckBox fontSize="XL" />
    </div>
</Preview>

### Props

<Props story="CheckBox" />
