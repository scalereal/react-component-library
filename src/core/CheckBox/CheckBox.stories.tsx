import { Meta, Story } from '@storybook/react/types-6-0';
import React from 'react';
import CheckBox from '.';
import { Props } from './types';

const SizeMap: Props['size'][] = ['XS', 'S', 'M', 'L', 'XL'];

export default {
    title: 'Components/Core/CheckBox',
    component: CheckBox,
    argTypes : {
        size : {
            control: {
                type: 'select',
                options: Object.values(SizeMap),
            },
        },
        label:{
            control:{type:'text'}
        },
        disabled:{
            control:{type:'boolean'}
        },
        hasError:{
            control: { type: 'boolean'}
        },
        errorText:{
            control:{type:'text'}
        },
        bgColor : {
            control:'color'
        },
        textColor: {
            control:'color'
        },
        hoverBgColor: {
            control:'color'
        },
        hoverTextColor: {
            control:'color'
        },
        hoverShadowColor: {
            control:'color'
        },
        hoverTextShadowColor: {
            control:'color'
        },
        hoverEffect:{ 
            control: { type: 'boolean'}  
        },
        
    }
} as Meta;

const Template: Story<Props> = ({...args }) => {
    // retrieves the appropriate icon passes it as a component prop
    return <CheckBox {...args} />;
};

export const Default = Template.bind({});


