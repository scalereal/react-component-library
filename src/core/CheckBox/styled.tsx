import styled from 'styled-components';
import { Props } from './types';

const mixin = (size:any) => {
    switch (size) {
        case "XS":
            return `width: 1.4rem;
            height: 1.4rem;
            label{

            }
            `;
        case "S":
            return `width: 1.8rem;
            height: 1.8rem;
            `;
        
        case "M":
            return `width: 2.2rem;
            height: 2.2rem;
            `;

        case "L":
            return `width: 2.6rem;
            height: 2.6rem;
            `;

        case "XL":
            return `width: 3rem;
            height: 3rem;
            `;
  
      default:
        return `border:1px solid red;`;
     }
 };

export const StyledCheckBoxInput = styled.input<Props>`
    margin:0.3rem;
    ${pr => pr.disabled === true?'pointer-events: none;opacity: 0.4;':''}
    ${pr=>  mixin(pr.size)}
    border-radius: 10%;
    text-align:center;
    background-color: ${pr => pr.bgColor || `#C4C4C4` };

    ${pr => pr.hoverEffect? `&:hover {
        background-color: ${pr.hoverBgColor || '#126AFA'};
        box-shadow: 0px 5px 8px ${pr.hoverShadowColor || '#004AD7' };
        }
        `:
        ''
    }
`;
export const StyledLabel = styled.label`
    display: inline-flex;
    cursor: pointer;
    padding: 0.8rem 0px;
    -webkit-box-align: center;
    align-items: center;
    -webkit-box-pack: end;
    justify-content: flex-end;
    span{
        cursor: inherit;
        color: inherit;
        padding: 0px 1.6rem;
    }
`;