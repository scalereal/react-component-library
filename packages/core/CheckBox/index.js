var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import React, { memo, useState } from 'react';
import Text from '../Text/Text';
import { CheckBoxWrapper, ErrorLabel, StyledLabel } from './styled';
// export type Ref = HTMLLabelElement;
var CheckBox = memo(React.forwardRef(function (props, ref) {
    var _a = props.size, size = _a === void 0 ? 'S' : _a, _b = props.label, label = _b === void 0 ? 'CheckBox' : _b, _c = props.value, value = _c === void 0 ? '' : _c, _d = props.errorText, errorText = _d === void 0 ? '' : _d, fontSize = props.fontSize, labelHoverEffect = props.labelHoverEffect, labelHoverColor = props.labelHoverColor, labelColor = props.labelColor, _e = props.checked, checked = _e === void 0 ? false : _e;
    var _f = useState(checked), check = _f[0], setCheck = _f[1];
    return (React.createElement(CheckBoxWrapper, __assign({ ref: ref }, props),
        React.createElement(ErrorLabel, { id: "error", htmlFor: "error" }, errorText),
        React.createElement(StyledLabel, __assign({ size: size }, props),
            React.createElement("input", { type: "checkbox", checked: checked ? checked : check, value: value === '' ? label : value, onChange: function () { return setCheck(!check); } }),
            React.createElement(Text, { textSize: fontSize, textColor: labelColor, hoverEffect: labelHoverEffect, hoverColor: labelHoverColor }, label))));
}));
CheckBox.displayName = "CheckBox";
CheckBox.defaultProps = {
    hoverEffect: false
};
export default CheckBox;
