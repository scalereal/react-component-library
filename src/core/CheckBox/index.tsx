import React from 'react';
import { ErrorLabel, StyledLabel } from './styled';
import { Props } from './types';

export type Ref = HTMLLabelElement;

let CheckBox = React.forwardRef<Ref, Props>( (props, ref) => {
    const {size='S', label='CheckBox',value='',errorText=''} = props;
  
    return ( 
        <div>
            <ErrorLabel id="error" htmlFor="error">{errorText}</ErrorLabel>
            <StyledLabel ref={ref} size={size} {...props} >
                <input type="checkbox" value={value===''?label:value}/>
                <span>{label}</span>
                {/* <Label>{label}</Label> */}
            </StyledLabel>
        </div>
        )
});

CheckBox.displayName="CheckBox";
CheckBox.defaultProps = {
    hoverEffect: false
};

export default CheckBox = React.memo(CheckBox)
