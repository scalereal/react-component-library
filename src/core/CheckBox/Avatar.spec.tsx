import { render } from '@testing-library/react';
import * as React from 'react';
import CheckBox from '.';

describe('Avatar component', () => {
    it('should render with default theme', () => {
        const { container } = render(<CheckBox/>);
        expect(container).toMatchSnapshot();
    });

});