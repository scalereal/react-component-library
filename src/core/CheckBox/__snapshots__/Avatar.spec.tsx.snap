// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component should render with default theme 1`] = `
<div>
  <label
    class="sc-gsTCUz fIhbEO"
  >
    <input
      class="sc-bdfBwQ ioezKP"
      type="checkbox"
    />
    <span>
      CheckBox
    </span>
  </label>
</div>
`;
